### DISCUSSION TITLE
DataIngestion PowerBI Report

### DISCUSSION THREAD

**Discussion by calVertical** [microsoft/finops-toolkit#1043] [Q&A] [answered]
Testing out the DataIngestion PowerBI report, I am receiving an error in the HubScopes query.

My scope appears as a record and on the step:

= if ADLS = null then null else Table.TransformColumns(Scope,{{"Scope", Text.Lower, type text}})

I receive the error:

Expression.Error: We cannot convert a value of type Record to type Text.
Details:
    Value=
        scope=/providers/Microsoft.Billing/billingAccounts/1234567
    Type=[Type]

In general, ingestion is working as expected with the current scope configuration.

Thanks in advance.

**Comment by flanakin** [comment, marked as answer]
Is this with 0.6? I'm aware of an issue there. Ingestion paths were changed but the Data ingestion report wasn't updated.

**Reply by flanakin** [reply]
@calVertical I spent some time on this yesterday and updated the latest Data ingestion report ([PBIX](https://github.com/microsoft/finops-toolkit/releases/latest/download/DataIngestion.pbix) / [PBIT](https://github.com/microsoft/finops-toolkit/releases/latest/download/DataIngestion.pbit)). Please download the latest. Thanks for reporting this and sorry about the inconvenience!

**Reply by calVertical** [reply]
Thanks for the quick response! I'm running on version 0.5.
I tested the new pbix(https://github.com/microsoft/finops-toolkit/releases/latest/download/DataIngestion.pbix) and still receive the same error on the HubScopes Lowercase step.

![Text Lower Error](https://github.com/user-attachments/assets/f9b242b9-6dec-4d20-9743-f228dcaf6e6a)

**Reply by jogudev** [reply]
I also got the same issue with a single billing account scope in v.0.6 of FinOps toolkit reports. I believe it is connected with having to adjust the settings.json to use a record for the scopes. Otherwise the config-pipeline in ADF did not work.

I fixed the issue by adding a step to the report that expands the record to a column (see content from advanced editor below):

<img width="877" alt="image" src="https://github.com/user-attachments/assets/c9daf794-d12e-456f-9ad2-0616397c0807">


I don't know if this is a long term solution, but it got me past the issue.

**Reply by flanakin** [reply]
We're about to ship 0.8 which. Let us know if you run into any issues there. I'll mark this resolved for now since we haven't heard back but please do create a new thread or issue if you run into problems or have any questions.
